clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java

```diff
@@ -127,12 +180,62 @@ private String artifactVersion(Artifact artifact) {
   }
 
   private CloneCommand addAuthentication(CloneCommand cloneCommand) {
-    // TODO(ethanfrogers): support github oauth token and ssh authentication
-    if (!StringUtils.isEmpty(username) && !StringUtils.isEmpty(password)) {
-      return cloneCommand.setCredentialsProvider(
-          new UsernamePasswordCredentialsProvider(username, password));
+    switch (authType) {
+      case HTTP:
+        return cloneCommand.setCredentialsProvider(
+            new UsernamePasswordCredentialsProvider(username, password));
+      case TOKEN:
+        return cloneCommand.setCredentialsProvider(
+            new UsernamePasswordCredentialsProvider(token, ""));
+      case SSH:
+        return configureSshAuth(cloneCommand);
+      default:
+        return cloneCommand;
     }
+  }
 
-    return cloneCommand;
+  private CloneCommand configureSshAuth(CloneCommand cloneCommand) {
+    SshSessionFactory sshSessionFactory =
+        new JschConfigSessionFactory() {
+          @Override
+          protected void configure(OpenSshConfig.Host hc, Session session) {
+            if (sshKnownHostsFilePath == null && sshTrustUnknownHosts) {
+              session.setConfig("StrictHostKeyChecking", "no");
+            }
+          }
+
+          @Override
+          protected JSch createDefaultJSch(FS fs) throws JSchException {
+            JSch defaultJSch = super.createDefaultJSch(fs);
+            defaultJSch.addIdentity(sshPrivateKeyFilePath, sshPrivateKeyPassphrase);
+
+            if (sshKnownHostsFilePath != null && sshTrustUnknownHosts) {
+              log.warn(
```

Comment:
It feels like this warning could be closer to where we actually ignore the value of `sshTrustUnknownHosts` (ie, in `confgure`).  (Or maybe that gets called too often to log...)